cdc_rsync/cdc_rsync_client.cc

```diff
@@ -396,10 +392,26 @@ absl::Status CdcRsyncClient::HandleServerOutput(const char* data) {
   }
 
   printer_.Print(stdout_data, false, Util::GetConsoleWidth());
-  if (!is_server_listening_) {
+  if (server_listen_port_ == 0) {
     server_output_.append(stdout_data);
-    is_server_listening_ =
-        server_output_.find("Server is listening") != std::string::npos;
+
+    // Parse port from "Port <n>: Server is listening".
+    size_t sil_pos = server_output_.find("Server is listening");
+    if (sil_pos != std::string::npos) {
+      // Search backwards until we find "Port ".
+      size_t port_pos = server_output_.rfind("Port ", sil_pos);
+      if (port_pos == std::string::npos) {
+        return MakeStatus("Failed to find 'Port' marker in server output '%s'",
+                          server_output_);
+      }
+      assert(sil_pos > port_pos);
+      server_listen_port_ =
+          atoi(server_output_.substr(port_pos + 5, sil_pos - port_pos).c_str());
```

Comment:
I assume 5 is an amount of chars in "Port ", can you please name this constant?

Reply:
Done.